wheels are not correctly built

When I then go to use mt2:
```
(coffea-dev) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % conda create -y --name test_env_5 numpy=1.23.5 python=3.9
(coffea-dev) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % conda activate test_env_5
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % pip install mt2
Collecting mt2
  Downloading mt2-1.2.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (7.6 kB)
Requirement already satisfied: numpy>=1.19.3 in /Users/lgray/miniforge3/envs/test_env_5/lib/python3.9/site-packages (from mt2) (1.23.5)
Downloading mt2-1.2.1-cp39-cp39-macosx_11_0_arm64.whl (22 kB)
Installing collected packages: mt2
Successfully installed mt2-1.2.1
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % python
Python 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:55:20) 
[Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
im>>> import mt2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'mt2'
>>> 
(test_env_5) lgray@Lindseys-MacBook-Pro dependent-repos-of-coffea % ls /Users/lgray/miniforge3/envs/test_env_5/lib/python3.9/site-packages
README.txt				distutils-precedence.pth		numpy-1.23.5.dist-info			pkg_resources				wheel
_distutils_hack				mt2-1.2.1.dist-info			pip					setuptools				wheel-0.44.0.dist-info
_mt2.cpython-39-darwin.so		numpy					pip-24.2.dist-info			setuptools-72.2.0-py3.12.egg-info
```

So something's not *quite* right in the way the wheels are being generated. The mt2 shared object library is being installed into the root of site-packages!

Okay, have a patch. FYI, also getting a couple of probably-easy-to-fix warnings:

```
In file included from src/main.cpp:10:
src/mt2_Lallyver2.h:536:34: warning: unused variable 'checkdeltasqsq' [-Wunused-variable]
                    const double checkdeltasqsq = checkdeltasq * checkdeltasq;
                                 ^
src/mt2_Lallyver2.h:564:34: warning: unused variable 'newdeltaLBsqsq' [-Wunused-variable]
                    const double newdeltaLBsqsq = newdeltaLBsq * newdeltaLBsq;
                                 ^
2 warnings generated.
```

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,0 +1,4 @@
+include *.rst
+include Makefile
+recursive-include examples *.ipynb
+recursive-include src *.h

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,8 +26,11 @@ requires-python = ">= 3.9"
 urls = { Homepage = "https://github.com/tpgillam/mt2" }
 
 [build-system]
-requires = ["setuptools>=61.0", "wheel", "numpy"]
+requires = ["setuptools>=61.0", "numpy"]
 build-backend = "setuptools.build_meta"
+
+[tool.setuptools.packages.find]
+include = ["mt2*"]
 
 [tool.setuptools.package-data]
 mt2 = ["src/*.h", "src/*.cpp"]

--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ __version__ = "1.2.1"
 setup(
     ext_modules=[
         Extension(
-            "mt2._mt2",
+            "_mt2",
             ["src/main.cpp"],
             define_macros=[
                 # Pass in the version info so we can expose it in the extension.

--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ __version__ = "1.2.1"
 setup(
     ext_modules=[
         Extension(
-            "_mt2",
+            "mt2._mt2",
             ["src/main.cpp"],
             define_macros=[
                 # Pass in the version info so we can expose it in the extension.
@@ -19,7 +19,7 @@ setup(
                 # citation information elsewhere.
                 ("DISABLE_COPYRIGHT_PRINTING", "1"),
             ],
-            include_dirs=[numpy.get_include()],
+            include_dirs=[numpy.get_include(), "src"],
             language="c++",
             extra_compile_args=["-std=c++11"],
         ),

